packages/ic-asset-certification/src/asset_router.rs

```diff
@@ -117,23 +117,39 @@ struct CertifiedAssetResponse<'a> {
 #[derive(Debug)]
 pub struct AssetRouter<'content> {
     tree: Rc<RefCell<HttpCertificationTree>>,
-    responses: HashMap<String, CertifiedAssetResponse<'content>>,
-    encoded_responses: HashMap<(String, String), CertifiedAssetResponse<'content>>,
-    fallback_responses: HashMap<String, CertifiedAssetResponse<'content>>,
-    encoded_fallback_responses: HashMap<(String, String), CertifiedAssetResponse<'content>>,
+    responses: HashMap<RequestKey, CertifiedAssetResponse<'content>>,
+    fallback_responses: HashMap<RequestKey, CertifiedAssetResponse<'content>>,
+}
+
+/// A key created from request data, to retrieve the corresponding response.
+#[derive(Debug, Eq, Hash, PartialEq)]
+pub struct RequestKey {
+    /// Path of the requested asset.
+    pub path: String,
+    /// The encoding of the assset.
+    pub encoding: Option<String>,
+}
+
+fn request_key(path: &str, encoding: Option<String>) -> RequestKey {
```

Comment:
It's only a style preference, but what do you think about implementing this as a constructor?

```rust
impl RequestKey {
  fn new(path: &str, encoding: Option<String>) -> Self {
    Self {
        path: path.to_string(),
        encoding,
    }
  }
}
```

Reply:
SGTM, I'll do that in the next PR that builds on top of this (so that I don't have to reconcile merge conflicts), ok?